### ISSUE TITLE
Some commit after 30 abr 2016 breaks preview animation in godot editor (Win764bit)

### ISSUE THREAD

**Issue by Ranoller** [kestrelm/Creature_Godot#7]
Godot version: 2.1.4 custom build.

Bulding with plugin code-state in 30 abr 2016 for windows do a plugin that acts equally that your youtube video: 
https://www.youtube.com/watch?v=afSObHcj9O8

Building with plugin code-state in 30 may 2016 make a plugin that do not allow to the user to preview animation in editor (only with play-preview, not directly in editor). Nothing changes in screen if you uses  "Anim Frame" making unable to do animation-frame directly in godot.

Any of this commits made the regression:

Commits on May 30, 2016
 @kestrelm
Updating SCons build for OSX
kestrelm committed on 30 May 2016
c116f8c  
 @kestrelm
Fixing animation blending bug
kestrelm committed on 30 May 2016
f75573f

**Comment by kestrelm** [comment]
Hello,

Both those changes do not touch any of the display code I think, especially the first one.
However, you can try commenting out the line in f75573f and see if it fixes the issue.

**Comment by Ranoller** [comment]
Yes  I did that... this is the problem, commenting (manager->SetAutoBlending(true);) fix the problem.

**Comment by kestrelm** [comment]
Ok, you can work with that for now. However, autoBlending must be set to true somewhere in order for the blending to work. The issue is I do not know through their API where to initialize that. LoadJson was the candidate to do it but if it conflicts with the preview system, then we can remove it for now. But I still think that needs to be called.

**Comment by kestrelm** [comment]
Actually, thinking about the problem a bit more, the correct fix is probably to only call the SetAutoBlending(true); when the code is running in the game, not in the editor preview. However, I am unable to find in their docs how to determine which "mode" the plugin is currently in. If anybody can shed some light on this issue that will be great.

**Comment by mrezai** [comment]
Maybe `Engine::get_singleton()->is_editor_hint()` is what you want

**Comment by kestrelm** [comment]
@mrezai  Hello all, this has been patched into the latest creature godot 3 code.

Cheers

**Comment by Ranoller** [comment]
Could be great to have this in godot2 module too... Godot3 is not in production-mode and there is middle-travel to arrival.

**Comment by Ranoller** [comment]
I put this for documenting:

If someone is using this script and want a comfortably workaround to this problem, can compile the godot editor with the module with line 123 commented and compile export templates uncommentig this line. with this workaround you have animations working in editor and exporting.
